src/bin/helper.rs

```diff
@@ -35,24 +40,27 @@ async fn main() -> Result<(), Box<dyn Error>> {
     let args = Args::parse();
     let _handle = args.logging.setup_logging();
 
-    // decide what protocol we're going to use here
-    let addr = SocketAddr::from(([127, 0, 0, 1], args.port.unwrap_or(0)));
-    let target = match args.scheme.as_str() {
-        "http" => BindTarget::Http(addr),
-        #[cfg(feature = "self-signed-certs")]
-        "https" => {
-            let config = raw_ipa::net::tls_config_from_self_signed_cert().await?;
-            BindTarget::Https(addr, config)
-        }
-        _ => {
-            panic!("\"{}\" protocol is not supported", args.scheme)
-        }
+    let peer_discovery_str =
+        std::fs::read_to_string("./peer_conf.toml").expect("unable to read file");
+    let peer_discovery =
+        discovery::conf::Conf::from_str(&peer_discovery_str).expect("unable to parse config file");
+    let gateway_config = GatewayConfig {
+        send_buffer_config: SendBufferConfig {
+            items_in_batch: 1,
+            batch_count: 40,
+        },
     };
+    let helper = HttpHelper::new(args.role, &peer_discovery, gateway_config);
+    let (addr, server_handle) = helper.bind().await;
+    let gateway = helper.query(QueryId).expect("unable to create gateway");
+    let step = Step::default().narrow(&PrssExchangeStep);
```

Comment:
narrowing should happen inside prss protocol imo

Reply:
yeah, that makes sense.